### PR TITLE
Fix `Pressables`'s interference with other gestures when nested

### LINKED ISSUE #2863: [Web] Nested Pressable inside a Swipeable captures press when swiping
### Description

Nesting a `Pressable` component inside a `Swipeable` works as expected on mobile:
1. tapping the `Pressable` content fires its `onPress` ✅ 
2. swiping the `Swipeable` works as expected and **does not** fire the `Pressable` `onPress` ✅ 

However, on web:
1. tapping the `Pressable` content fires its `onPress` ✅ 
2. swiping the `Swipeable` works as expected **but also** fires the `Pressable` `onPress` ❌ 

https://github.com/software-mansion/react-native-gesture-handler/assets/144704010/4968e4d2-b226-49b2-b229-a5c19fcfacb6

https://github.com/software-mansion/react-native-gesture-handler/assets/144704010/db8cead7-b42a-431a-a79a-6466850bc94d

### Steps to reproduce

1. Clone linked repository for minimum reproducible example
2. Run app on web
3. Swipe the swipeable row
- **Expected**: `onPress` isn't fired
- **Actual**: `onPress` is fired, alert is shown

### Snack or a link to a repository

https://github.com/milan-digiuseppe-level/swipeable-nested-pressable-mvp

### Gesture Handler version

2.14.1

### React Native version

0.73.6

### Platforms

iOS, Web

### JavaScript runtime

None

### Workflow

Expo managed workflow

### Architecture

None

### Build type

Debug mode

### Device

None

### Device model

_No response_

### Acknowledgements

Yes

### PR DESCRIPTION
**Pull request by latekvo**
## Description

`LongPress` was found to be using excessively high value for it's `minDuration` and `maxDistance` configs.

While `maxDistance` config caused no issues, `minDuration` config caused an instantaneous activation.
This behaviour stems from `setTimeout` handling values up to `2^31-1` while `MAX_SAFE_INTEGER` is much higher than that. 
This unintended activation blocked some nested gestures underneath the `Pressable` from activating on `web`.

By replacing `Number.MAX_SAFE_INTEGER` with `2 ** 31 - 1` which represents the largest possible 32-bit integer, those issues were resolved.

Found while investigating [this issue](https://github.com/software-mansion/react-native-gesture-handler/issues/2863). 

Both issues - the one reported by @milan-digiuseppe-level and the one described in this PR might be related, but it's not immidiately clear as I couldn't replicate the former one.

closes #2863

## Test plan

- open the `EmptyExample` in both an editor and a `web` browser
- replace the default exported function with the attached code
- without this fix, only 2 of the 4 `Swipeables` can be moved by their nested `Pressable`, with this fix, all of them can be swiped

Before | After
------ | ------
 <video src="https://github.com/user-attachments/assets/069076dc-c624-4c24-bf9b-f4bbc7a0ad79"> | <video src="https://github.com/user-attachments/assets/8cc507c2-04b5-44e9-a8ad-8fcc3f0b6cb2">

## Attached code
```js
/* eslint-disable no-alert */
import React from 'react';
import { Text, Animated, StyleSheet, View, Pressable } from 'react-native';

import {
  Pressable as GHPressable,
  Swipeable,
  GestureHandlerRootView,
} from 'react-native-gesture-handler';
import ReanimatedSwipeable from 'react-native-gesture-handler/ReanimatedSwipeable';
import Reanimated, {
  SharedValue,
  useAnimatedStyle,
} from 'react-native-reanimated';

function LeftAction(prog: SharedValue<number>, drag: SharedValue<number>) {
  const styleAnimation = useAnimatedStyle(() => {
    console.log('[R] showLeftProgress:', prog.value);
    console.log('[R] appliedTranslation:', drag.value);

    return {
      transform: [{ translateX: drag.value - 50 }],
    };
  });

  return (
    <Reanimated.View style={styleAnimation}>
      <Text style={styles.leftAction}>Text</Text>
    </Reanimated.View>
  );
}

function RightAction(prog: SharedValue<number>, drag: SharedValue<number>) {
  const styleAnimation = useAnimatedStyle(() => {
    console.log('[R] showRightProgress:', prog.value);
    console.log('[R] appliedTranslation:', drag.value);

    return {
      transform: [{ translateX: drag.value + 50 }],
    };
  });

  return (
    <Reanimated.View style={styleAnimation}>
      <Text style={styles.rightAction}>Text</Text>
    </Reanimated.View>
  );
}

function LegacyLeftAction(prog: any, drag: any) {
  prog.addListener((value: any) => {
    console.log('[L] showLeftProgress:', value.value);
  });
  drag.addListener((value: any) => {
    console.log('[L] appliedTranslation:', value.value);
  });

  const trans = Animated.subtract(drag, 50);

  return (
    <Animated.Text
      style={[
        styles.leftAction,
        {
          transform: [{ translateX: trans }],
        },
      ]}>
      Text
    </Animated.Text>
  );
}

function LegacyRightAction(prog: any, drag: any) {
  prog.addListener((value: any) => {
    console.log('[L] showRightProgress:', value.value);
  });
  drag.addListener((value: any) => {
    console.log('[L] appliedTranslation:', value.value);
  });

  const trans = Animated.add(drag, 50);

  return (
    <Animated.Text
      style={[
        styles.rightAction,
        {
          transform: [{ translateX: trans }],
        },
      ]}>
      Text
    </Animated.Text>
  );
}

export default function Example() {
  return (
    <GestureHandlerRootView>
      <View style={styles.separator} />

      <ReanimatedSwipeable
        containerStyle={styles.swipeable}
        friction={2}
        leftThreshold={80}
        enableTrackpadTwoFingerGesture
        rightThreshold={56}
        renderLeftActions={LeftAction}
        renderRightActions={RightAction}>
        <Pressable onPress={() => alert('pressed!')} style={styles.pressable}>
          <Text>[new] with RN pressable</Text>
        </Pressable>
      </ReanimatedSwipeable>

      <View style={styles.separator} />

      <ReanimatedSwipeable
        containerStyle={styles.swipeable}
        friction={2}
        leftThreshold={80}
        enableTrackpadTwoFingerGesture
        rightThreshold={56}
        renderLeftActions={LeftAction}
        renderRightActions={RightAction}>
        <GHPressable onPress={() => alert('pressed!')} style={styles.pressable}>
          <Text>[new] with GH pressable</Text>
        </GHPressable>
      </ReanimatedSwipeable>

      <View style={styles.separator} />

      <Swipeable
        containerStyle={styles.swipeable}
        friction={2}
        leftThreshold={80}
        enableTrackpadTwoFingerGesture
        rightThreshold={56}
        renderLeftActions={LegacyLeftAction}
        renderRightActions={LegacyRightAction}>
        <Pressable onPress={() => alert('pressed!')} style={styles.pressable}>
          <Text>[Legacy] with RN pressable</Text>
        </Pressable>
      </Swipeable>

      <View style={styles.separator} />

      <Swipeable
        containerStyle={styles.swipeable}
        friction={2}
        leftThreshold={80}
        enableTrackpadTwoFingerGesture
        rightThreshold={56}
        renderLeftActions={LegacyLeftAction}
        renderRightActions={LegacyRightAction}>
        <GHPressable onPress={() => alert('pressed!')} style={styles.pressable}>
          <Text>[Legacy] with GH pressable</Text>
        </GHPressable>
      </Swipeable>

      <View style={styles.separator} />
    </GestureHandlerRootView>
  );
}

const styles = StyleSheet.create({
  leftAction: { width: 46, height: 56, backgroundColor: 'crimson' },
  rightAction: { width: 46, height: 56, backgroundColor: 'purple' },
  separator: {
    width: '100%',
    borderTopWidth: 1,
  },
  swipeable: {
    height: 56,
    backgroundColor: 'papayawhip',
    alignItems: 'center',
  },
  pressable: {
    padding: 20,
    width: 200,
    height: 56,
    backgroundColor: 'pink',
    alignItems: 'center',
  },
});

```

### CODE REVIEW

**Comment by latekvo** [comment]
After further testing, it appears that on `iOS`, when `Pressable` is nested inside a native `Scroll`, only every second press is registered. I'll address this issue asap.

**Comment by j-piasecki** [comment]
Doesn't changing `LongPress` to `Manual` change the behavior on native? I believe we were talking about it and `Manual` was causing other gestures to misbehave.

**Comment by latekvo** [comment]
> Doesn't changing `LongPress` to `Manual` change the behavior on native? I believe we were talking about it and `Manual` was causing other gestures to misbehave. - @j-piasecki 

It does!
But today I discovered that by adding `.manualActivation(true)` to the `Manual` gesture, most of those issues can be alleviated. The only one that I have to fix after this change is the one I mentioned in the comment above.

> Also, the title is a bit confusing as it suggests changes to Swipeable component.

I'll fix the title.

File: src/utils.ts [hunk 1]
```diff
@@ -98,3 +98,5 @@ export function deepEqual(obj1: any, obj2: any) {
 
   return true;
 }
+
+export const maxSafeTimeout = 2 ** 31 - 1;
```
Comment (m-bert) [inline src/utils.ts:102]: I'd change this name to represent what it actually is (for example in C there is `INT_MAX`)

**Review by m-bert — CHANGES_REQUESTED** [review]

**Comment by m-bert** [comment]
Also, please update PR description 😅

File: src/utils.ts (hunk 1 above)
Comment (latekvo) [reply src/utils.ts:102]: I think `INT_MAX` overlaps with existing `MAX_SAFE_INTEGER` without providing context for why it exists.
What about `MAX_SAFE_INT32` or `MAX_SAFE_TIMEOUT`?
The former signals it's the max 32 bit value while the latter that it's safe for timeouts.

Comment (m-bert) [reply src/utils.ts:102]: I'd like to avoid using `timeout`. If we were to use it later, it would be confusing. `INT32_MAX` sounds ok.

**Review by m-bert — APPROVED** [review]

**Review by j-piasecki — APPROVED** [review]

**Pull request merged**